pkg/credentialprovider/shim/shim.go

```diff
@@ -302,20 +289,23 @@ func (p *shimProvider) getCredentialsForImage(
 				)
 		}
 
-		cacheDuration := prov.DefaultCacheDuration()
-		if v1beta1Resp.CacheDuration != nil {
-			cacheDuration = v1beta1Resp.CacheDuration.Duration
+		for k, v := range v1beta1Resp.Auth {
+			if matched, _ := urlglobber.URLsMatchStr(k, img); !matched {
+				continue
+			}
+
+			if len(longestMatchedURL) < len(k) || longestMatchedURL < k {
```

Comment:
Just a note for my future self: `longestMatchedURL < k` compares the strings lexicographically